zerver/tests/tests.py

```diff
@@ -2132,7 +2132,7 @@ def test_open_realm_logic(self):
         mit_realm.save()
         self.assertTrue(completely_open(mit_realm.domain))
         self.assertEquals(get_unique_open_realm(), None)
-        with self.settings(SYSTEM_ONLY_REALMS={"zulip.com"}):
+        with self.settings(SYSTEM_ONLY_REALMS={"zulip.com", "analytics.test"}):
```

Comment:
Aha!  It's because you put this:
```
    default_realm = Realm.objects.create(domain='analytics.test', name='Realm Test',                 
                                         date_created=TIME_ZERO - 2*DAY)                             
```
in the class definition, so it runs at import time, not during the test runtime (and thus not part of the transaction that gets aborted).  Just move it to a `setUp` method in the class (and then you probably don't even need a teardown).  Creating a realm takes <1ms, I don't think doing it once per test will make any difference :)